scipy/optimize/minpack.py

```diff
@@ -449,23 +449,29 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
         return (retval[0], info)
 
 
-def _wrap_func(func, xdata, ydata, weights):
-    if weights is None:
+def _wrap_func(func, xdata, ydata, T):
+    if T is None:
         def func_wrapped(params):
             return func(xdata, *params) - ydata
+    elif T.shape == xdata.shape:
```

Comment:
Sort of minor, but why don't check the dimensionality? Like if `T.ndim == 1`, it feels simpler.